core/src/main/java/tc/oc/pgm/community/modules/FreezeMatchModule.java

```diff
@@ -0,0 +1,349 @@
+      Location old = event.getFrom();
+      old.setPitch(event.getTo().getPitch());
+      old.setYaw(event.getTo().getYaw());
+      event.setTo(old);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+  public void onVehicleMove(final VehicleMoveEvent event) {
+    if (!event.getVehicle().isEmpty() && freeze.isFrozen(event.getVehicle().getPassenger())) {
+      event.getVehicle().setVelocity(new Vector(0, 0, 0));
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onVehicleEnter(final VehicleEnterEvent event) {
+    if (freeze.isFrozen(event.getEntered())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onVehicleExit(final VehicleExitEvent event) {
+    if (freeze.isFrozen(event.getExited())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onBlockBreak(final BlockBreakEvent event) {
+    if (freeze.isFrozen(event.getPlayer())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onBlockPlace(final BlockPlaceEvent event) {
+    if (freeze.isFrozen(event.getPlayer())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onBucketFill(final PlayerBucketFillEvent event) {
+    if (freeze.isFrozen(event.getPlayer())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onBucketEmpty(final PlayerBucketEmptyEvent event) {
+    if (freeze.isFrozen(event.getPlayer())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW) // ignoreCancelled doesn't seem to work well here
+  public void onPlayerInteract(final PlayerInteractEvent event) {
+    if (freeze.isFrozen(event.getPlayer())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onInventoryClick(final InventoryClickEvent event) {
+    if (event.getWhoClicked() instanceof Player) {
+      if (freeze.isFrozen(event.getWhoClicked())) {
+        event.setCancelled(true);
+      }
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onPlayerDropItem(final PlayerDropItemEvent event) {
+    if (freeze.isFrozen(event.getPlayer())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onEntityDamge(final EntityDamageByEntityEvent event) {
+    if (freeze.isFrozen(event.getDamager())) {
+      event.setCancelled(true);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void onVehicleDamage(final VehicleDamageEvent event) {
```

Comment:
This should maybe be placed together with the other VeichleEvents? And why not just cancel all `VehicleEvent`s? :p

Reply:
You can’t cancel VehicleEvent xD